Unable to lint MDX components in `2.0.0-next.2`

### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

2.0.0-next.2

### Link to runnable example

_No response_

### Steps to reproduce

```js
module.exports = {
  overrides: [
    {
      files: '*.md{,x}',
      parser: 'eslint-mdx',
      processor: 'mdx/remark',
      plugins: ['mdx'],
      settings: {
        'mdx/code-blocks': true
      }
    },
    {
      files: ['*.md{,x}', '*.{j,t}s{,x}'],
      extends: 'plugin:react/recommended'
    }
  ]
}
```

```mdx
# Hello world

\```js
<Foo />
\```

<Foo />

\```jsx
<Foo />
\```
```

### Expected behavior

```
/Users/dimitri/Desktop/mdx-repro/test.mdx
   7:1  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
   7:2  error  'Foo' is not defined                     react/jsx-no-undef
   4:1  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
   4:2  error  'Foo' is not defined                     react/jsx-no-undef
  10:1  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
  10:2  error  'Foo' is not defined                     react/jsx-no-undef

✖ 6 problems (6 errors, 0 warnings)
```

### Actual behavior

```
   4:1  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
   4:2  error  'Foo' is not defined                     react/jsx-no-undef
  10:1  error  'React' must be in scope when using JSX  react/react-in-jsx-scope
  10:2  error  'Foo' is not defined                     react/jsx-no-undef

✖ 4 problems (4 errors, 0 warnings)
```

### Runtime

_No response_

### Package manager

_No response_

### OS

_No response_

### Build and bundle tools

_No response_

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     "@types/eslint": "^8.4.2",
     "@types/eslint-plugin-markdown": "^2.0.0",
     "@types/jest": "^27.5.0",
-    "@types/node": "^17.0.31",
+    "@types/node": "^17.0.32",
     "@types/react": "^18.0.9",
     "@types/unist": "^2.0.6",
     "lerna": "^4.0.0",

--- a/test/fixtures.test.ts
+++ b/test/fixtures.test.ts
@@ -22,6 +22,9 @@ const getCli = (lintCodeBlocks = false) =>
           version: 'detect',
         },
       },
+      rules: {
+        'unicorn/template-indent': 0,
+      },
       overrides: lintCodeBlocks
         ? [
             {

--- a/test/fixtures/391.mdx
+++ b/test/fixtures/391.mdx
@@ -1,0 +1,11 @@
+# Hello world
+
+```js
+<Foo />
+```
+
+<Foo />
+
+```jsx
+<Foo />
+```

